extension/src/setup/index.ts

```diff
@@ -220,21 +220,20 @@ export class Setup
     return this.sendDataToWebview()
   }
 
-  public async sendDataToWebview() {
-    const projectInitialized = this.hasRoots()
-    const hasData = this.getHasData()
+  public shouldBeShown() {
+    return !this.cliCompatible || !this.hasRoots() || !this.getHasData()
```

Comment:
[Q] Does this mean some complexity can be removed from the extension's `package.json`? 
i.e around this code:

```
    "viewsWelcome": [
      {
        "view": "dvc.views.webviews",
        "contents": "[Show Experiments](command:dvc.showExperiments)\n[Show Plots](command:dvc.showPlots)\n[Show Experiments and Plots](command:dvc.showExperimentsAndPlots)",
        "when": "dvc.commands.available && dvc.project.available && dvc.project.hasData"
      },
      {
        "view": "dvc.views.webviews",
        "contents": "[Show Experiments](command:dvc.showSetup)\n[Show Plots](command:dvc.showSetup)\n[Show Experiments and Plots](command:dvc.showSetup)",
        "when": "!dvc.commands.available || !dvc.project.available || !dvc.project.hasData"
      },
```

Reply:
You're right! I'll remove that.